Packs/AWS-Enrichment-Remediation/Scripts/AWSAccountHierarchy/AWSAccountHierarchy.py

```diff
@@ -0,0 +1,151 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from typing import Dict, Any
+import traceback
+
+
+""" STANDALONE FUNCTION """
+
+
+def lookup(parent_obj: str, level: int, instance_to_use: str) -> tuple[str, dict]:
+    """
+    Lookup information on a organization unit (ou) or root object. Unless the current lookup
+    is a root object, it returns parent object to lookup and results of current lookup.
+    Args:
+        parent_obj (str): root/ou object in number format.
+        level (int): the current level that the object is (ascending).
+        instance_to_use (str): what integration instance to use.
+
+    Returns:
+        str: parent object to look up next or "NONE" for error handling and "stop" if no parent.
+        dict: dictionary of id, name and level of the lookup object.
+
+    """
+    temp: Dict[str, str] = {}
+    try:
+        if "ou-" in parent_obj:
+            ou_info = execute_command(
+                "aws-org-organization-unit-get",
+                {"organization_unit_id": parent_obj, "using": instance_to_use},
+            )
+            if not ou_info:
+                return "NONE", temp
+            temp["level"] = str(level)
+            temp["id"] = ou_info.get("Id", "")
+            temp["name"] = ou_info.get("Name", "")
+            temp["arn"] = ou_info.get("Arn", "")
+            ou_parent = execute_command(
+                "aws-org-parent-list",
+                {"child_id": parent_obj, "using": instance_to_use},
+            )
+            if not ou_parent:
+                return "NONE", temp
+            next_one = ou_parent[0].get("Id", "")
+        elif "r-" in parent_obj:
+            next_one = "stop"
+            root_info = execute_command("aws-org-root-list", {"using": instance_to_use})
+            if not root_info:
+                return "NONE", temp
+            root = root_info.get("AWS.Organizations.Root(val.Id && val.Id == obj.Id)")[
+                0
+            ]
+            temp["level"] = str(level)
+            temp["id"] = root.get("Id", "")
+            temp["name"] = root.get("Name", "")
+            temp["arn"] = root.get("Arn", "")
+        else:
+            raise ValueError("unexpected object type")
+    except TypeError:
+        return "NONE", temp
+    else:
+        return next_one, temp
+
+
+""" COMMAND FUNCTION """
+
+
+def aws_account_heirarchy(args: Dict[str, Any]) -> CommandResults:
+    """
+    Determine AWS account hierarchy by looking up parent objects until the root level is reached.
+    Args:
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        list[CommandResults]: outputs, readable outputs and raw response for XSOAR.
+
+    """
+
+    account_id = args.get("account_id")
+
+    if not account_id:
+        raise ValueError("account_id not specified")
```

Comment:
```suggestion
```

If **account_id** is required then it is safe to assume it was given.

Reply:
this automation is modeled off this one: https://github.com/demisto/content/blob/94481c3deb734cbaf15c91e07cd47b8db21f9bd3/Packs/GCP-Enrichment-Remediation/Scripts/GCPProjectHierarchy/GCPProjectHierarchy.py#L63

prefer to keep this code block